### PR TITLE
feat(account): add menu for managing API credentials

### PR DESCRIPTION
**Pull request by antleblanc**
⚠️ Please **do not merge** this pull request.

ref:
- #178
- ovh-ux/ovh-api-services#75 

needed:
- ovh-ux/ovh-ui-angular#189

### CODE REVIEW

File: client/app/account/user/credentials/delete/user-credentials-delete.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div data-ng-controller="UserAccount.controllers.credentials.delete as $ctrl">
+
+    <div data-wizard
+         data-wizard-on-cancel="resetAction"
+         data-wizard-on-finish="deleteCredential"
+         data-wizard-title="i18n.user_credentials_delete_modal_title">
+
+        <div data-wizard-step>
+
+            <p data-ng-bind-html="tr('user_credentials_delete_modal_step1_question', [$ctrl.credential.name])"></p>
```
Comment (jleveugle) [inline client/app/account/user/credentials/delete/user-credentials-delete.html]: tr? Should be $translate now :)

File: client/app/account/user/credentials/user-credentials.routes.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+angular
+    .module("UserAccount")
+    .config(["$stateProvider", function ($stateProvider) {
```
Comment (jleveugle) [inline client/app/account/user/credentials/user-credentials.routes.js]: No need to inject $stateProvider before the function, grunt ng-annotate task do it for you :)

File: client/app/account/user/credentials/user-credentials.service.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+angular.module("UserAccount.services")
+    .service("UseraccountCredentialsService", class UseraccountCredentialsService {
```
Comment (jleveugle) [inline client/app/account/user/credentials/user-credentials.service.js]: It should be in ovh-api-services :)

File: client/app/account/user/credentials/user-credentials.service.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        constructor (OvhHttp) {
+            this.OvhHttp = OvhHttp;
+        }
+
+        getCredentials () {
+            return this.OvhHttp.get("/me/api/credential", {
```
Comment (jleveugle) [inline client/app/account/user/credentials/user-credentials.service.js]: you should use a variable for the pattern: "/me/api/credential" and use it across your methods :)

File: client/app/account/user/credentials/delete/user-credentials-delete.html (hunk 1 above)
Comment (antleblanc) [reply client/app/account/user/credentials/delete/user-credentials-delete.html]: Yes I need to rebase on `develop` branch.

File: client/app/account/user/credentials/user-credentials.service.js (hunk 3 above)
Comment (antleblanc) [reply client/app/account/user/credentials/user-credentials.service.js]: You're right! I'll have to merge and release this [pull request](https://github.com/ovh-ux/ovh-api-services/pull/84) first.

File: client/app/account/user/credentials/user-credentials.service.js (hunk 4 above)
Comment (antleblanc) [reply client/app/account/user/credentials/user-credentials.service.js]: `OvhApiMe` will be used.

File: client/app/account/user/credentials/user-credentials.html [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+        <!-- CREDENTIALS LIST -->
+        <oui-datagrid data-rows-loader="$ctrl.loadDatagridCredentials($config)"
+                      data-row-loader="$ctrl.transformItem($row)">
+            <oui-column data-title=":: i18n.user_credentials_name"
+                        data-property="name">
+            </oui-column>
+            <oui-column data-title=":: i18n.user_credentials_desc">
+                <span data-ng-if="$row.description.length > 20"
+                      data-ng-bind="($row.description | limitTo:'15') + '…'"
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/user-credentials.html]: 👍

File: client/app/account/user/credentials/user-credentials.controller.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+angular.module("UserAccount.controllers")
+    .controller("UserAccount.controllers.credentials", class UserAccountCredentialsController {
+
+        constructor ($q, $scope, Alerter, UseraccountCredentialsService) {
+            this.$q = $q;
+            this.$scope = $scope;
+            this.Alerter = Alerter;
+            this.UseraccountCredentialsService = UseraccountCredentialsService;
+
+            this.$scope.$on("useraccount.credentials.refresh", () => {
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/user-credentials.controller.js]: Move to `$onInit` ?

File: client/app/account/user/credentials/user-credentials.controller.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+        /**
+         * Get the list of your Api Credentials IDs.
+         * @return {Promise}
+         */
+        getCredentials () {
+            return this.UseraccountCredentialsService.getCredentials()
+                .then((credentialsIds) => {
+                    this.credentialsIds = credentialsIds;
+                    return credentialsIds;
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/user-credentials.controller.js]: Is this needed ? You already have `this.credentialsIds` at this point

File: client/app/account/user/credentials/user-credentials.controller.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                        data: part.map((id) => ({ id })),
+                        meta: {
+                            totalCount: this.credentialsIds.length
+                        }
+                    };
+                });
+        }
+
+        transformItem (credential) {
+            return this.UseraccountCredentialsService
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/user-credentials.controller.js]:
```js
return this.UseraccountCredentialsService
	.getCredential(credential.id)
    .then((credentialInfo) => this.UseraccountCredentialsService
    	.getApplication(credentialInfo.credentialId)
        .then((applicationInfo) => _.assign(credentialInfo, applicationInfo)));
```

to

```js
return this.UseraccountCredentialsService
	.getCredential(credential.id)
    .then((credentialInfo) => this.UseraccountCredentialsService.getApplication(credentialInfo.credentialId))
    .then((applicationInfo) => _.assign(credentialInfo, applicationInfo));
```

File: client/app/account/user/credentials/read/user-credentials-read.html [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+                            data-i18n-static="user_credentials_permissions_path">
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr data-ng-repeat="rule in $ctrl.credential.rules track by $index">
+                        <td>
+                            <span class="label"
+                                  data-ng-class="{
+                                      'label-danger': rule.method === 'DELETE',
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/read/user-credentials-read.html]: Why is `DELETE` considered a "danger" ? I understand it's only a matter of color but it's a bit weird, was this CX validated ?

File: client/app/account/user/credentials/delete/user-credentials-delete.controller.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+"use strict";
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/delete/user-credentials-delete.controller.js]: No need for this if you use a `class`

File: client/app/account/user/credentials/delete/user-credentials-delete.controller.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+angular.module("UserAccount.controllers")
+    .controller("UserAccount.controllers.credentials.delete", class UserAccountCredentialsDeleteController {
+
+        constructor ($scope, UseraccountCredentialsService, Alerter) {
+            this.$scope = $scope;
+            this.UseraccountCredentialsService = UseraccountCredentialsService;
+            this.Alerter = Alerter;
+            this.credential = $scope.currentActionData;
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/delete/user-credentials-delete.controller.js]: You can move this to `$onInit`

File: client/app/account/user/credentials/read/user-credentials-read.controller.js [hunk 12]
```diff
@@ -0,0 +1,7 @@
+angular.module("UserAccount.controllers")
+    .controller("UserAccount.controllers.credentials.read", class UserAccountCredentialsReadController {
+        constructor ($scope) {
+            this.$scope = $scope;
+            this.credential = $scope.currentActionData;
```
Comment (FredericEspiau) [inline client/app/account/user/credentials/read/user-credentials-read.controller.js]: You can move this to `$onInit`

**Comment by antleblanc** [comment]
I close this PR due to inactivity. I plan to re-open it in a near future.

**Pull request closed without merging**